client/src/modules/authorization/components/SignInForm.jsx

```diff
@@ -0,0 +1,185 @@
+import PendingButton from 'common/components/PendingButton';
+import { AbortPromiseException } from 'common/exceptions/AbortPromiseException';
+import { makeAbortablePromise } from 'common/utils/helpers';
+import { UnauthorizedException } from 'common/exceptions/UnauthorizedException';
+import { IntlPropType } from 'common/constants/propTypes';
+
+class SignInForm extends PureComponent {
+  pendingPromise = null;
+
+  state = {
+    email: '',
+    isEmailValid: false,
+    isFormValid: false,
+    isPasswordValid: false,
+    password: '',
+    pending: false,
+    signUpErrorId: ''
+  };
+
+  componentDidUpdate() {
+    this.isFormValid();
+  }
+
+  componentWillUnmount() {
+    if (this.pendingPromise) {
+      this.pendingPromise.abort();
+    }
+  }
+
+  onEmailChange = (email, isValid) => {
+    const { signUpErrorId } = this.state;
+    const error = isValid ? '' : signUpErrorId;
+    this.setState({
+      email,
+      isEmailValid: isValid,
+      signUpErrorId: error
+    });
+  };
+
+  onPasswordChange = (password, isValid) => {
+    const { signUpErrorId } = this.state;
+    const error = isValid ? '' : signUpErrorId;
+    this.setState(
+      {
+        password,
+        isPasswordValid: isValid,
+        signUpErrorId: error
+      },
+      this.comparePasswords
+    );
+  };
+
+  emailValidator = value => {
+    const { isEmail } = validator;
+
+    if (!isEmail(value)) {
+      return 'authorization.input.email.invalid';
+    }
+
+    return '';
+  };
+
+  isFormValid = () => {
+    const { isEmailValid, isPasswordValid, signUpErrorId } = this.state;
+
+    return this.setState({
+      isFormValid: isEmailValid && isPasswordValid && !signUpErrorId
+    });
+  };
+
+  handleSignIn = () => {
+    const { email, password } = this.state;
+    const { signIn } = this.props;
+
+    this.setState({ pending: true });
+
+    this.pendingPromise = makeAbortablePromise(signIn(email, password));
+
+    return this.pendingPromise.promise
+      .then(() => this.setState({ pending: true }))
+      .catch(err => {
+        if (!(err instanceof AbortPromiseException)) {
+          const newState = { pending: false };
+
+          if (err instanceof UnauthorizedException) {
+            newState.signUpErrorId =
+              'authorization.actions.sign-in.invalid-credentials';
+          } else {
+            newState.signUpErrorId = 'common.something-went-wrong';
+          }
+
+          this.setState(newState);
+        }
+      });
+  };
+
+  renderSignUpError = () => {
+    const { signInErrorId } = this.state;
+    const {
+      intl: { formatMessage }
+    } = this.props;
+    const message = `${formatMessage({ id: signInErrorId })} ${formatMessage({
+      id: 'common.try-again'
+    })}`;
+
+    return <p className="sign-in-form__error">{message}</p>;
+  };
+
+  render() {
+    const { isFormValid, pending, signUpErrorId } = this.state;
+    const { onCancel } = this.props;
+    const hasSignUpFailed = signUpErrorId !== '';
```

Comment:
you can also check the length 

const hasSignUpFailed = signUpErrorId.length > 0